### PR TITLE
browser: android fixes v2

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes APP-1342
Fixes APP-1436
Fixes APP-1340

also added i18n where missing, fixed some small android style issues

### CODE REVIEW

File: src/components/DappBrowser/BrowserTab.tsx [hunk 1]
```diff
@@ -529,7 +538,6 @@ const styles = StyleSheet.create({
     overflow: 'hidden',
     top: 0,
     width: DEVICE_WIDTH,
-    zIndex: 50000,
```
Comment (walmat) [inline src/components/DappBrowser/BrowserTab.tsx:532]: over 9000

File: src/components/DappBrowser/search-input/SearchInput.tsx [hunk 2]
```diff
@@ -74,85 +75,104 @@ const TheeDotMenu = function TheeDotMenu({
...
 
-    [activeTabInfo, canGoBack, canGoForward, isFavorite]
+    return {
+      menuTitle: '',
+      menuItems,
+    };
+  }, [canGoBack, canGoForward, isFavorite, tabUrl]);
+
+  const onPressMenuItem = useCallback(
+    async ({ nativeEvent: { actionKey } }: { nativeEvent: { actionKey: 'share' | 'favorite' | 'back' | 'forward' } }) => {
```
Comment (walmat) [inline src/components/DappBrowser/search-input/SearchInput.tsx:129]: would prefer these all to be typed properly going forward.

import { OnPressMenuItemEventObject } from 'react-native-ios-context-menu';

File: src/components/DappBrowser/search-input/SearchInput.tsx [hunk 3]
```diff
@@ -74,85 +75,104 @@ const TheeDotMenu = function TheeDotMenu({
...
-    }
-  };
+  const onPressAndroid = useCallback(() => {
+    showActionSheetWithOptions(
+      {
+        ...{ cancelButtonIndex: menuConfig.menuItems.length - 1 },
+        options: menuConfig.menuItems.map(item => item?.actionTitle),
+      },
+      (buttonIndex: number) => {
+        onPressMenuItem({ nativeEvent: { actionKey: menuConfig.menuItems[buttonIndex]?.actionKey as any } });
```
Comment (walmat) [inline src/components/DappBrowser/search-input/SearchInput.tsx:152]: same here, use the proper type:
```
import { OnPressMenuItemEventObject } from 'react-native-ios-context-menu';
```

**Review by walmat — APPROVED** [review]
small nits but code looks good.

File: src/components/DappBrowser/search-input/SearchInput.tsx [hunk 4]
```diff
@@ -74,85 +75,104 @@ const TheeDotMenu = function TheeDotMenu({
...
+      haptics.selection();
+      if (actionKey === 'favorite') {
+        handleFavoritePress();
+      } else if (actionKey === 'back') {
+        goBack();
+      } else if (actionKey === 'forward') {
+        goForward();
+      } else {
+        const url = activeTabInfo.value.url;
+        if (url) handleShareUrl(url);
```
Comment (walmat) [inline src/components/DappBrowser/search-input/SearchInput.tsx:139]: i think this clause will be hit on android if they cancel the action sheet.

Comment (walmat) [reply src/components/DappBrowser/search-input/SearchInput.tsx:139]: only change that's needed. Others are nits @skylarbarrera

Comment (skylarbarrera) [reply src/components/DappBrowser/search-input/SearchInput.tsx:139]: im gonna merge this in and handle in my next PR, nice catch

**Pull request merged**